bridge/github/import.go

```diff
@@ -43,40 +46,60 @@ func (gi *githubImporter) Init(conf core.Configuration) error {
 
 // ImportAll iterate over all the configured repository issues and ensure the creation of the
 // missing issues / timeline items / edits / label events ...
-func (gi *githubImporter) ImportAll(repo *cache.RepoCache, since time.Time) error {
+func (gi *githubImporter) ImportAll(ctx context.Context, repo *cache.RepoCache, since time.Time) (<-chan core.ImportResult, error) {
 	gi.iterator = NewIterator(gi.conf[keyOwner], gi.conf[keyProject], gi.conf[keyToken], since)
+	out := make(chan core.ImportResult)
+	gi.out = out
 
-	// Loop over all matching issues
-	for gi.iterator.NextIssue() {
-		issue := gi.iterator.IssueValue()
-		fmt.Printf("importing issue: %v\n", issue.Title)
+	go func() {
+		defer close(gi.out)
 
-		// create issue
-		b, err := gi.ensureIssue(repo, issue)
-		if err != nil {
-			return fmt.Errorf("issue creation: %v", err)
-		}
+		// Loop over all matching issues
+		for gi.iterator.NextIssue() {
+			select {
+			case <-ctx.Done():
```

Comment:
So essentially, you allow to stop the process between each bugs. That's a clear improvement but I think it can be done better. 

What you could do is:
- pass the context down up to the functions doing the actual network calls (which means those will actually be cancelled instead of waiting). When that happen, everything will return early with (I suppose) an error (the one given by the context).
- in this function, check the error. If it's a context cancelled error, you can safely commit the bug to keep what was successfully imported before the cancel.
- in this function,

Reply:
Note: this is because you don't need to cancel anything done locally. It's fast and will take a few milliseconds worst case. What you really need to cancel is everything where you need to really wait or that can hang (so in this case, the network calls).

Reply:
> pass the context down up to the functions doing the actual network calls (which means those will actually be cancelled instead of waiting). When that happen, everything will return early with (I suppose) an error (the one given by the context).

Is it OK to change the `iterator` methods signatures to use `context`  ? We'll have something like `NextIssue(ctx context.Context)`, `NextComment(ctx context.Context)` , `NextTimelineItem(ctx context.Context)`  ...

What if one of these functions doesn't have to make an api call (already cached some items), should they be checking the `ctx.Err()` at the start of each call ?

Reply:
Maybe pass the context when creating the iterator ?

Reply:
Done 👍

Reply:
hmmm, if you handle `<-ctx.Done()` in the iterator, you can't have it here as well. Unless I'm mistaken, there is a single value sent to this channel so you essentially have a race condition.

Reply:
Correct ! it is called twice in the same function